numba/lowering.py

```diff
@@ -405,6 +405,25 @@ def lower_binop(self, resty, expr):
         return self.context.cast(self.builder, res, signature.return_type,
                                  resty)
 
+
+    def lower_inplace_binop(self, resty, expr):
+        lhs = expr.lhs
+        rhs = expr.rhs
+        lty = self.typeof(lhs.name)
+        rty = self.typeof(rhs.name)
+        lhs = self.loadvar(lhs.name)
+        rhs = self.loadvar(rhs.name)
+        # Get function
+        expr.fn = expr.fn + '='
```

Comment:
@pitrou Here IR produced by numba for the following function using inplace operator.
```python
@jit(nopython=True)
def add_5(x):
    x += 5
```
```
-----------------------------IR DUMP: inplace_add_5-----------------------------
label 0:
    x = arg(0, name=x)                       ['x']
    $const0.2 = const(int, 5)                ['$const0.2']
    $0.3 = inplace_binop(rhs=$const0.2, lhs=x, fn=+) ['$0.3', '$const0.2', 'x']
    del x                                    []
    del $const0.2                            []
    x.1 = $0.3                               ['$0.3', 'x.1']
    del x.1                                  []
    del $0.3                                 []
    $const0.4 = const(NoneType, None)        ['$const0.4']
    $0.5 = cast(value=$const0.4)             ['$0.5', '$const0.4']
    del $const0.4                            []
    return $0.5                              ['$0.5']
```

notice that `x += 5` is replaced by `$0.3 = inplace_binop(rhs=$const0.2, lhs=x, fn=+) ['$0.3', '$const0.2', 'x']`, since the `fn` here is `+` and I mutated the IR to make it `+=` so that mapping produced by `_op_map` works. I guess the work around would be change the code producing IR so that `fn` will be `+=`.